File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < numbers.length; i++) {
            int comp = target - numbers[i];

            if (map.containsKey(comp)) {
                return new int[]{map.get(comp)+1, i+1};
            }

            map.put(numbers[i], i);
        }

        return new int[]{-1, -1};
    }
}

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.php
class Solution {
    /**
     * @param Integer[] $numbers
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($numbers, $target) {
        $left = 0;
        $right = count($numbers) - 1;

        while ($left < $right) {
            $sum = $numbers[$left] + $numbers[$right];

            if ($sum == $target) {
                return [$left + 1, $right + 1]; // Adjust indices to be 1-based
            } elseif ($sum < $target) {
                $left++;
            } else {
                $right--;
            }
        }

        // Return an empty array if no solution is found
        return [];
    }
}
